src/main/java/org/wildfly/security/auth/provider/SimpleMapBackedSecurityRealm.java

```diff
@@ -118,5 +121,17 @@ public CredentialSupport getCredentialSupport(Class<?> credentialType) {
             final Password password = map.get(principal);
             return credentialType.isInstance(password) ? credentialType.cast(password) : null;
         }
+
+        public VerificationResult verifyCredential(final Object credential) {
```

Comment:
Yeah I was trying to work out the combinations, like "can be verified but not acquired", or "can maybe be verified or maybe acquired" or so on like that.  Also taking into account that some combinations are inherently conceptually invalid, for example if you can acquire a credential, there's no reason why you can't verify one.